Client/mods/deathmatch/logic/CClientVectorGraphicDisplay.h

```diff
@@ -0,0 +1,64 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+class CClientVectorGraphicDisplay;
+
+#pragma once
+
+#include "CClientDisplay.h"
+#include "CClientDisplayManager.h"
+
+class CClientVectorGraphicDisplay : public CClientDisplay
+{
+    friend class CClientDisplayManager;
+
+public:
+    CClientVectorGraphicDisplay(CClientDisplayManager* pDisplayManager, CClientVectorGraphic* pVectorGraphic, int ID = DISPLAY_VECTORGRAPHIC);
+    ~CClientVectorGraphicDisplay() = default;
+
+    eDisplayType GetType() { return DISPLAY_VECTORGRAPHIC; }
+
+    void SetPosition(const CVector& vecPosition);
+
+    void SetColor(const SColor color);
+    void SetColorAlpha(unsigned char ucAlpha);
+    void SetShadowAlpha(unsigned char ucShadowAlpha);
+
+    float GetScale() { return m_fScale; };
+    void  SetScale(float fScale);
+
+    unsigned long GetFormat() { return m_ulFormat; };
+    void          SetFormat(unsigned long ulFormat);
+
+    void SetVisible(bool bVisible);
+
+    void Render();
+
+    void UpdateTexture();
+    void ClearTexture();
+
+    static void SetGlobalScale(float fScale) { m_fGlobalScale = fScale; }
+
+    const bool IsCleared() { return m_bIsCleared; }
+    const bool HasUpdated() { return m_bHasUpdated; }
+
+    void Update();
+private:
+    CClientVectorGraphic* m_pVectorGraphic;
+    SString m_strCaption;
+    float   m_fScale;
+
+    unsigned long m_ulFormat;
+    unsigned char m_ucShadowAlpha;
+
+    static float m_fGlobalScale;
```

Comment:
Do you even use it? What is it for?

Reply:
Copied from another class which inherits `CClientDisplay` (which I used as reference) - probably not needed so will remove.

Reply:
Actually, I will probably remove this class entirely and integrate the logic directly into `CClientVectorGraphic` instead. 

I based this on a similar class purely so I could use existing render pipelines, but looking at it now it doesn't seem necessary. I'll take a proper look later on.

Reply:
Decided to just remove unnecessary definitions here.